### PR TITLE
Marking as obsolete all changes in public API

### PR DESCRIPTION
**Pull request by WojcikMike**
Connects to #87

Related to doco PR: https://github.com/Particular/docs.particular.net/pull/1437

### CODE REVIEW

**Comment by WojcikMike** [comment]
@Particular/azure-storage-queues-maintainers  I need some help on Obsolete messages, please have a look at obsolete.cs file as a lot of types were removed or made internal. I usually don't have good explanation on why, so I put a generic text saying:
"This class served only internal purposes without providing any extensibility point and as such was removed from teh public API, please refer to the documentation for more information."

I will put into documentation information about changing in our public API with regards about keeping public what only is needed for users

File: src/Tests/API/APIApprovals.ApproveAzureStorageQueueTransport.approved.txt [hunk 1]
```diff
@@ -1,6 +1,41 @@
 ﻿namespace NServiceBus.Azure.Transports.WindowsAzureStorageQueues
 {
     
+    [System.ObsoleteAttribute("This class served only internal purposes without providing any extensibility poin" +
```
Comment (SeanFeldman) [inline src/Tests/API/APIApprovals.ApproveAzureStorageQueueTransport.approved.txt]: > removed from teh public API
type "teh" => "the"

File: src/Tests/API/APIApprovals.ApproveAzureStorageQueueTransport.approved.txt [hunk 2]
```diff
@@ -1,6 +1,41 @@
...
+        "t and as such was removed from teh public API, please refer to the documentation" +
```
Comment (SeanFeldman) [inline src/Tests/API/APIApprovals.ApproveAzureStorageQueueTransport.approved.txt]: How about turning 

> , please refer to the documentation...

into  ". For more information, refer to the documentation."?

File: src/Transport/AzureMessageQueueCreator.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿namespace NServiceBus.Azure.Transports.WindowsAzureStorageQueues
+﻿namespace NServiceBus.Azure.Transports.WindowsAzureStorageQueues.Internals
```
Comment (SeanFeldman) [inline src/Transport/AzureMessageQueueCreator.cs]: Have to admit I'm not a big fan of ".Internals" namespace.
Would rather see a calls marked with accessibility modifier, which seems to be a no-go for the core folks.

Comment (WojcikMike) [inline src/Transport/AzureMessageQueueCreator.cs]: Yeah I don't like it either, I asked number of people how to obsolete gracefully class that was changed to internal and the options were:
* rename the leftover class to something else, and keep the original type with obsolete
* move the leftover class to other namespace, and keep the original type with obsolete

Both of those solutions have pros and cons. What is more i don't have any good name for the namespace. So I am happy to adjust this change to something else that will allow us to achieve the goal

File: src/Tests/API/APIApprovals.ApproveAzureStorageQueueTransport.approved.txt (hunk 1 above)
Comment (WojcikMike) [inline src/Tests/API/APIApprovals.ApproveAzureStorageQueueTransport.approved.txt]: Done.

File: src/Transport/AzureMessageQueueCreator.cs (hunk 3 above)
Comment (SeanFeldman) [inline src/Transport/AzureMessageQueueCreator.cs]: Discussed with @WojcikMike and we decided to apply the following:

* Keep obsoleted stuff in the original namespace.
* Anything that is public and needs to be easily discovered by users in `NServiceBus` namespace.
* The rest in Project.Namespace (fe: `NServiceBus.AzureStorageQueues`)

**Comment by WojcikMike** [comment]
After discussing with @SeanFeldman yesterday I introduced all outlined things.

I also addressed things that came out during conversation with @SimonCropp, @Scooletz and @andreasohlund 

Please review and merge

File: src/Transport/Dispatcher.cs [hunk 4]
```diff
@@ -15,7 +16,7 @@
 
     class Dispatcher : IDispatchMessages
     {
-        public Dispatcher(CreateQueueClients createQueueClients, MessageWrapperSerializer messageSerializer, QueueAddressGenerator addressGenerator, AzureStorageAddressingSettings addressing)
+        public Dispatcher(AzureStorageQueues.CreateQueueClients createQueueClients, MessageWrapperSerializer messageSerializer, QueueAddressGenerator addressGenerator, AzureStorageAddressingSettings addressing)
```
Comment (Scooletz) [inline src/Transport/Dispatcher.cs]: Why `Dispatcher` requires a full name of `CreateQueueClients`?

**Comment by Scooletz** [comment]
I'm :+1: with all the changes above. @SeanFeldman could you review it and merge if everything is ok? You had much more involvement in this, hence imo you'd be the one accepting it.

**Pull request merged**